### PR TITLE
Use preconfigured filters correctly in Analyze API

### PR DESCRIPTION
**Pull request by romseygeek**
When a named token filter or char filter is passed as part of an Analyze API 
request with no index, we currently try and build the relevant filter using no
index settings.  However, this can miss cases where there is a pre-configured
filter defined in the analysis registry.  One example here is the elision filter, which
has a pre-configured version built with the french elision set; when used as part
of normal analysis, this preconfigured set is used, but when used as part of the
Analyze API we end up with NPEs because it tries to instantiate the filter with
no index settings.

This commit changes the Analyze API to check for pre-configured filters in the case
that the request has no index defined, and is using a name rather than a custom 
definition for a filter.

Relates to #43002

### CODE REVIEW

**Comment by romseygeek** [comment]
This is an interesting failure: the pre-configured EdgeNGramTokenizer uses the lucene default values for min and max, which are both 1; however, the elasticsearch `EdgeNGramTokenizerFactory` instead uses default values from `NGramTokenizer`, giving a min of 1 and a max of 2.  This is a pretty clear bug - settings of `tokenizer=edge_ngram` and `tokenizer.type=edge_ngram` give you different configurations.

**Review by cbuescher — COMMENTED** [review]
The main change in AnalysisRegistry makes sense to me looking at the underlying issue. When reading the test I was wondering if it would be better (and/or easier) to start testing AnalysisRegistry#buildCustomAnalyzer() more directly in AnalysisRegistryTests instead of going through TransportAnalyzeActionTests. Not a heard requirement for this PR but maybe there are things that could be pulled out or added easily.

**Comment by romseygeek** [comment]
> When reading the test I was wondering if it would be better (and/or easier) to start testing AnalysisRegistry#buildCustomAnalyzer() more directly in AnalysisRegistryTests instead of going through TransportAnalyzeActionTests

+1, that makes sense, I'll open a follow-up issue

File: modules/analysis-common/src/test/java/org/elasticsearch/analysis/common/WordDelimiterGraphTokenFilterFactoryTests.java [hunk 1]
```diff
@@ -107,4 +117,51 @@ public void testAdjustingOffsets() throws IOException {
...
+            Settings settings = Settings.builder()
+                .put(Environment.PATH_HOME_SETTING.getKey(), createTempDir().toString())
+                .build();
+            Settings indexSettings = Settings.builder()
+                .put(IndexMetaData.SETTING_VERSION_CREATED,
+                    VersionUtils.randomVersionBetween(random(), Version.V_7_0_0, VersionUtils.getPreviousVersion(Version.V_7_3_0)))
+                .put("index.analysis.analyzer.my_analyzer.tokenizer", "standard")
+                .putList("index.analysis.analyzer.my_analyzer.filter", "word_delimiter_graph")
+                .build();
+            IndexSettings idxSettings = IndexSettingsModule.newIndexSettings("index", indexSettings);
```
Comment (cbuescher) [inline modules/analysis-common/src/test/java/org/elasticsearch/analysis/common/WordDelimiterGraphTokenFilterFactoryTests.java]: same here

**Review by cbuescher — APPROVED** [review]
Left a minor comment around testing, feel free to disregard if you don't like the idea. Rest LGTM

**Pull request merged**